[Tanveer Singh] iP

tanveersingh10 opened the pull request:
# Bot Cristiano

Cristiano provides full support for task management. It is:
- easy to learn
- super convenient 
- interactive user interface 
- Features the great Cristiano Ronaldo since Messi wasn't available

## All you need to do is,
Install from [here](https://github.com/tanveersingh10/ip/releases/tag/A-Jar)
Add your todos / deadlines / events
Let it manage your tasks for you 🔥💯:+1: 

Features:
- [x] managing todos
- [x] managing events and deadlines 
- [x] saving all your important information
- [x] fun user experience
- [ ] reminders: in the future!

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

tanveersingh10 committed:
Level 1

tanveersingh10 committed:
Level 0

tanveersingh10 committed:
Level 1

tanveersingh10 committed:
Level 2

tanveersingh10 committed:
Level 3

tanveersingh10 committed:
Level 3

tanveersingh10 committed:
Level 4

tanveersingh10 committed:
A-TextUiTesting

tanveersingh10 committed:
Level 5

tanveersingh10 committed:
Level 6

tanveersingh10 committed:
updated testing

tanveersingh10 committed:
new features

tanveersingh10 committed:
Level 7

tanveersingh10 committed:
Level 7

tanveersingh10 committed:
Level 8

tanveersingh10 committed:
Merge branch 'branch-Level-8'

tanveersingh10 committed:
OOP added

tanveersingh10 committed:
Merge branch 'A-MoreOOP'

tanveersingh10 committed:
Merge branch 'gradle'

tanveersingh10 committed:
fixed packaging

tanveersingh10 committed:
A-JUnit

tanveersingh10 committed:
Merge branch 'A-JUnit'

tanveersingh10 committed:
added docs

Seonlo99 commented on src/main/java/duke/main/UI.java:
```diff
@@ -0,0 +1,73 @@
+        printLine();
+        System.out.println("Here are the duke.tasks in your list:");
+        for (int i = 0; i < numOfTasks; i++) {
+            System.out.println((i + 1) + ". " + list.get(i));
```
I think it might be better to get the number of tasks by list.size() instead of passing in an additional parameter.

Seonlo99 commented on src/main/java/duke/main/UI.java:
```diff
@@ -0,0 +1,73 @@
+        printLine();
+    }
+    public void addTask(String taskName, int numOfTasks) {
+        printLine();
```
I think it will be better if there is a print message function that includes the print line so that printLine() will  not be needed at every command.

Seonlo99 commented on src/main/java/duke/exceptions/DukeException.java:
```diff
@@ -0,0 +1,11 @@
+package duke.exceptions;
```
I think it's really great of you to implement a variety of Exception classes.

Seonlo99 commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,199 @@
+                        case "list":
+                            taskManager.list();
+                            break;
+                        case "mark":
+                        case "unmark":
+                            handleMarking(input, taskManager);
+                            break;
```
I think you might have forgotten to handle the "mark" case.

Seonlo99 commented on src/main/java/duke/main/Duke.java:
```diff
@@ -0,0 +1,199 @@
+         * @param taskManager
+         * @throws InvalidArgumentException
+         */
+        private void handleMarking(String input, TaskManager taskManager) throws DukeException {
+            String[] words = input.split(" ");
+            try {
+                int index = Integer.parseInt(words[1]);
+                if (words[0].equals("mark")) {
+                    taskManager.mark(index);
+                } else if (words[0].equals("unmark")) {
+                    taskManager.unmark(index);
+                }
+                updateStorage();
+            } catch (NumberFormatException e) {
+                throw new InvalidArgumentException("Please enter a numerical index!");
+            } catch (StorageException e) {
+                throw new StorageException("duke.main.Storage Error");
+            }
+        }
+
+        private void updateStorage() throws StorageException {
+            this.storage.saveData(taskManager);
+        }
```
I think it might be better if all the command handling are contained inside the TaskManager class.

Seonlo99 commented on src/main/java/duke/main/Storage.java:
```diff
@@ -0,0 +1,117 @@
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.io.*;
```
I think it will be better if you just import the classes you need instead of the whole package.

Seonlo99 reviewed (commented):
LGTM. just some nits to fix.

tanveersingh10 committed:
added find functionality

tanveersingh10 committed:
Merge branch 'Level-9'

tanveersingh10 committed:
updated build.gradle

tanveersingh10 committed:
updated ui

tanveersingh10 committed:
Connected UI to application

tanveersingh10 committed:
Merge branch 'Level-10'

dinde2004 commented on src/main/java/duke/main/TaskManager.java:
```diff
@@ -0,0 +1,131 @@
+     */
+    public void mark(int index) throws InvalidArgumentException {
+        if (index > numOfTasks) {
+            throw new InvalidArgumentException("I'm sorry but that task does not exist. There are only " + numOfTasks + "duke.tasks.");
```
I think that one line should not contain more than 120 characters. You can consider line wrapping.

dinde2004 commented on src/main/java/duke/main/TaskManager.java:
```diff
@@ -0,0 +1,131 @@
+     */
+    public void unmark(int index) throws InvalidArgumentException {
+        if (index > numOfTasks) {
+            throw new InvalidArgumentException("I'm sorry but that task does not exist. There are only " + numOfTasks + " duke.tasks.");
```
I think that one line should not contain more than 120 characters. You can consider line wrapping.

dinde2004 commented on src/main/java/duke/main/TaskManager.java:
```diff
@@ -0,0 +1,131 @@
+     */
+    public void delete(int index) throws InvalidArgumentException {
+        if (index > numOfTasks) {
+            throw new InvalidArgumentException("I'm sorry but that task does not exist. There are only " + numOfTasks + " duke.tasks.");
```
I think that one line should not contain more than 120 characters. You can consider line wrapping.

dinde2004 commented on src/main/java/duke/tasks/Task.java:
```diff
@@ -0,0 +1,83 @@
+ */
+public class Task {
+    protected String taskName;
+    protected boolean done;
```
I think you should name boolean sound like boolean, for example: isDone instead of done.

dinde2004 reviewed (commented):
Overall, despite some coding standard violations like naming and length of line, your code is very good. I really like the way you use a wide variety of exceptions.

tanveersingh10 committed:
new release

tanveersingh10 committed:
there are no assertions made in the program.

hence, we are unable to check if the programmer's assumptions
are correct.

Let's add assert statements in the relevant places to validate
the programmer's assumptions

tanveersingh10 committed:
Merge pull request #1 from tanveersingh10/A-Assertions

Assertions

tanveersingh10 committed:
some code does not follow single layer of abstraction

needs to change in order to adhere to good code quality

let's refactor our code to follow good practice

tanveersingh10 committed:
added CI

tanveersingh10 committed:
Merge branch 'master' of github.com:tanveersingh10/ip

tanveersingh10 committed:
Merge pull request #2 from tanveersingh10/A-CodeQuality

some code does not follow single layer of abstraction

tanveersingh10 committed:
feature to add a note to tasks

tanveersingh10 committed:
Merge branch 'D-Notes'

tanveersingh10 committed:
more tests

tanveersingh10 committed:
Merge branch 'A-MoreTesting'

tanveersingh10 committed:
added more bot personality

tanveersingh10 committed:
removed presaved data

tanveersingh10 committed:
user guide

tanveersingh10 committed:
updated user guide

tanveersingh10 committed:
removed dead code

tanveersingh10 committed:
updated variable names

tanveersingh10 committed:
fixed nits

tanveersingh10 committed:
removed presaved tasks

tanveersingh10 committed:
added UI screenshot
